crates/parser/src/parse/source.rs

```diff
@@ -0,0 +1,22 @@
+use crate::codegen::SyntaxKind;
+use crate::Parser;
+
+use super::statement::statement;
+use super::statement_start::is_at_stmt_start;
+
+pub fn source(parser: &mut Parser) {
+    parser.start_node(SyntaxKind::SourceFile);
+
+    while !parser.eof() {
+        match is_at_stmt_start(parser) {
+            Some(stmt) => {
+                statement(parser, stmt);
```

Comment:
custom parsers can be added here later, and `statement` would just be the fallback.